app/controllers/products_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# Products CRUD
+class ProductsController < ApplicationController
+  before_action :authenticate_user!,except: %i[show]
+  before_action :load_product, only: %i[edit update destroy]
+  before_action :load_seller_products, only: %i[index]
+
+  def index; end
+
+  def new
+    @product = Product.new
+  end
+
+  def show
+    @product = Product.find_by(slug: params[:id])
+    @comments = @product.comments
+    @comment = Comment.new
+    render 'index/product'
+  end
+
+  def create
+    @product = current_user.products.new(product_params)
+    if @product.save
+      flash[:success] = 'New Product created'
+      redirect_to products_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @product.update(product_params)
+      flash[:success] = 'Product updated'
+      redirect_to products_path
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @product.destroy
```

Comment:
what will happen if the transaction fails?  this will always show the user a success message that is incorrect